### PR TITLE
Avi credentials added sub-options

### PR DESCRIPTION
**Pull request by chaitanyaavi**
##### SUMMARY

Avi credentials added sub-options

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- module_utils/network/avi/ansible_utils.py
- plugins/doc_fragments/avi.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by grastogi23 — APPROVED** [review]

**Review by grastogi23 — COMMENTED** [review]
+shipit

File: lib/ansible/module_utils/network/avi/ansible_utils.py [hunk 1]
```diff
@@ -544,13 +544,28 @@ def avi_common_argument_spec():
     Returns common arguments for all Avi modules
     :return: dict
     """
+    credentials_spec = dict(
+        controller=dict(default=os.environ.get('AVI_CONTROLLER', '')),
```
Comment (Qalthos) [inline lib/ansible/module_utils/network/avi/ansible_utils.py]: I think you should instead use `fallback` in these cases. See https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#fallback

For example:
```suggestion
        controller=dict(fallback=(env_fallback, ['AVI_CONTROLLER'])),
```

(Requires `from ansible.module_utils.basic import env_fallback` above)

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Review by grastogi23 — APPROVED** [review]
+shipit

**Comment by grastogi23** [comment]
@Qalthos Thanks for your suggestion. This was very helpful and we would rewire some more modules with similar approach.

**Pull request merged**